### ISSUE TITLE
Could addplot (somehow) add a 'non'-twoway plot object to an existing twoway graph?

### ISSUE THREAD

**Issue by ericmelse** [benjann/addplot#1]
Dear Ben,

I am most likely asking for something that is impossible, but could addplot (somehow) add a 'non'-twoway plot object to an existing twoway graph?

What I try to do is to first draw a twoway scatteri graph like:
```
tw scatteri 4 1 -6 1 -6 4 4 4 , graphregion(fcolor(white) ilcolor(none) lcolor(none) margin(l-3 r+.5 t-1 b-3)) ///
	mc(none) yscale(r(-6(1)4)) xscale(r(.5(1)4.5)) ytitle("Theta (s.d.)", size(*1.1)) xtitle(" ")	///
	name(tst, replace)
```
which looks like an empty grid and that is what I want to use.

Next, I want to add four separate graphs that use the (categorical) x-value of this plot, respectively 1, 2, 3 and 4, and use the y-scale of this plot, with values in between -6 and 4.
So, basically it is the twoway graph's y and x-axes that should determine where these (user contributed package created) non-twoway graphs are to be added onto the plot area.

You might think that using - graph combine - code be usefull and indeed that will produce something that looks like a result but there are awfull complications that distort the scaling of the combined graphs (both on the y-axis and the spread along the x-axis). That makes it an untrustworthy visualization.

I also searched for an alternative by trying to edit the above 'empty grid plot' so that the original proportion along the y-axis (upside of the plot) is kept but 'crop' the whole graph area from the right side close to the y-axis, leaving only a couple of grid line dots. Should that be possible, then indeed I could work with - graph combine - code to assemble that cropped graph with the four separate graphs into a single plot and not distort the scale etc.
But, I do not see if such 'cropping' is possible with Stata graphs, using the Graph Editor or any other (undocumented) functionality that allows for coding (replication).

So, hence my effort to get a solution using addplot.

Finally, as a last (and maybe best) alternative: do you know what does it take for a user contributed package to generate a 'twoway plot object'?
Does that require coding or something else that is proprietary to Stat Corp.?
Or would it require some special way of coding the ado file to become compatible with twoway and thus with addplot?

Maybe you can offer me some advice or help with this.

Best,
Eric Melse

**Comment by benjann** [comment]
Hi Eric,
I am not sure whether I fully understand, but I believe it is absolutely necessary that a graph command is provided as a twoway plot type for `addplot` to be able to process it. That is, the graph command must be made available as `twoway myplot ...`, assuming `myplot` is the name of your command. In principle, it is possible to define such commands, but I cannot really recall the details. All I remember is that class programming has to be used (see `help classman`). Many years ago I wrote such a program for `kdens` (`ssc describe kdens`). Example:
```
tw scatteri 4 1 -6 1 -6 4 4 4 , graphregion(fcolor(white) ilcolor(none) lcolor(none) margin(l-3 r+.5 t-1 b-3)) ///
    mc(none) yscale(r(-6(1)4)) xscale(r(.5(1)4.5)) ytitle("Theta (s.d.)", size(*1.1)) xtitle(" ")	///
    name(tst, replace)
sysuse auto, clear
addplot: kdens headroom, norescaling
```
If I remember right I just used official `twoway_kdensity_parse.class` and adapted it to `kdens`. So this is probably the easiest way to go: take an official `twoway_..._parse.class` and modify it to work with your command. ("easiest" does not necessarily mean "easy"; personally, I found the class stuff rather confusing...)
ben
